k8s-tf-cluster/01-datasources.tf

```diff
@@ -0,0 +1,13 @@
+/*
+This file defines the data sources for the GCP project
+*/
+
+data "google_project" "project" {
+  project_id = var.project
+}
+
+data "google_container_engine_versions" "us-central1" {
+  project        = data.google_project.project.id
```

Comment:
Based on the [doc](https://www.terraform.io/docs/providers/google/d/google_project.html), there is no attribute `id`. Is terraform trying to extract this information from the GCP API ?

Reply:
Docs are inaccurate. Using a data resource for this confirms that A) the project exists and B) that you have access to that project.

This value is computed to the GCP project ID (e.g. `k8s-infra-dev-cluster-turnup`).